integr8ly/integreatly-operator
# MGDAPI-4895 - resolve go.opentelemetry.io/proto/otlp replaces

# Issue link

https://issues.redhat.com/browse/MGDAPI-4895

# What

* Resolve go.opentelemetry.io/proto/otlp replaces
  * Remove use of `github.com/operator-framework/operator-registry/pkg/lib/bundle` as it causing a protobuf namespace conflict issue until it is resolved
      *  https://github.com/operator-framework/operator-registry/issues/1114
* Update gosec to v2.16.0
# Verification steps



* Passing prow check should be sufficient to ensure no breaking changes except for `gosec`
* `make gosec` is failing due to the image was previously using v2.11.0 with the following error:
```
[gosec] 2023/06/20 14:11:43 Checking file: /Users/chfan/dev/integreatly-operator/pkg/resources/constants/subscriptionConstants.go
2023/06/20 14:11:45 Unexpected package creation during export data loading
make: *** [gosec] Error 1
```
Updating to v2.16.0 will resolve this but this PR needs to be merged first since the base image is still using the old version. You can install the updated version locally and running gosec locally to ensure there is no issues: 
```
# Might need to delete current version if hitting the same issue
go install github.com/securego/gosec/v2/cmd/gosec@v2.16.0
make vendor/fix
make gosec
```
